Examples/Cocoa/Tests/Objective_CTests/ObjcEqualityTests.swift

```diff
@@ -0,0 +1,464 @@
+//
+//  ObjcEqualityTests.swift
+//  CoreTests
+//
+//  Created by Greg Bolsinga on 6/10/19.
+//
+
+import XCTest
+
+@testable import Objective_C
+
+class ObjcEqualityTests: XCTestCase {
+//    func testEquality_arrayProp ()
+//    {
+//        let t = 0
+//        let builder = EverythingBuilder()
+//        builder.arrayProp = t
+//        let e1 = builder.build()
+//        let e2 = builder.build()
+//        XCTAssertEqual(e1.arrayProp, t)
+//        XCTAssertEqual(e2.arrayProp, t)
+//        XCTAssertEqual(e1, e2)
+//        XCTAssertEqual(e1.hash,e2.hash)
```

Comment:
If I uncomment this one, and make the `let t` assignment accurate, I get unusual swift errors I could use some help with.

Reply:
Just tested this myself, the only thing missing here is additional type information so the compiler can ensure that the equality check is between two objects that conform to `Equatable`. Here is a diff of the changes you'll need here:

```swift
❯ git diff
 @testable import Objective_C

 class ObjcEqualityTests: XCTestCase {
-//    func testEquality_arrayProp ()
-//    {
-//        let t = 0
-//        let builder = EverythingBuilder()
-//        builder.arrayProp = t
-//        let e1 = builder.build()
-//        let e2 = builder.build()
-//        XCTAssertEqual(e1.arrayProp, t)
-//        XCTAssertEqual(e2.arrayProp, t)
-//        XCTAssertEqual(e1, e2)
-//        XCTAssertEqual(e1.hash,e2.hash)
-//    }
+    func testEquality_arrayProp ()
+    {
+        let t: [String] = []
+        let builder = EverythingBuilder()
+        builder.arrayProp = t
+        let e1 = builder.build()
+        let e2 = builder.build()
+        XCTAssertEqual(e1.arrayProp as? [String], t)
+        XCTAssertEqual(e2.arrayProp as? [String], t)
+        XCTAssertEqual(e1, e2)
+        XCTAssertEqual(e1.hash,e2.hash)
+    }

```

Reply:
Thanks for the tip, @rahul-malik ! I've updated all the tests. Each property in `Everything` is individually unit tested for equality. The `booleanProp` ones fail with that bad merge from my other PR. Thanks again!